fog/view/server/src/router_request_handler.rs

```diff
@@ -262,9 +281,26 @@ async fn query_shard(
     request: &MultiViewStoreQueryRequest,
     shard: Shard,
 ) -> Result<(Shard, MultiViewStoreQueryResponse), RouterServerError> {
-    let client_unary_receiver = shard.grpc_client.multi_view_store_query_async(request)?;
-    let response = client_unary_receiver.await?;
+    let start_time = Instant::now();
+    let subdomain = shard.uri.subdomain().unwrap_or("");
+    let histogram_observe = |status: &str| {
+        let status = status.chars().take(10).collect::<String>();
```

Comment:
Why only the first 10 characters?

Reply:
Generally metric labels are supposed to be pretty short. It's generally not recommended to put full error messages in a metric label. I'm not sure what the errors will actually look like, but was hoping for it to give an idea of the type of error that we could then perhaps use logs to correlate.

Now that we generally have things working though, maybe it's overkill and I'll just put static "error" wdyt?

Reply:
Makes sense regarding not putting the entire error string there, however I think we would either want to custom-tailor a solution or increase it from 10.

I believe the error types being formatted are [grpcio::Error](https://docs.rs/grpcio/latest/src/grpcio/error.rs.html#37), and the exact RpcFailure error might be interesting but `RpcFailure: ` is already 12 characters so we won't see the actual error code. So maybe increasing to 20 characters is going to be acceptable.